src/authentication/web3-login/container.test.tsx

```diff
@@ -17,11 +19,12 @@ describe('Web3LoginContainer', () => {
     it('should map state to props', () => {
       const state = {
         login: { loading: true, errors: ['error'] },
+        web3: { status: ConnectionStatus.Connected },
       };
 
       const props = Container.mapState(state as any);
 
-      expect(props).toEqual({ error: 'error', isConnecting: true });
+      expect(props).toEqual({ error: 'error', isConnecting: true, isWalletConnected: true });
```

Comment:
`isConnecting` is related to the `login` state, e.g. is the user currently signing the authentication message or waiting for a response from the server. `isWalletConnected` is related to the `web3` state, e.g. there is a wallet connected, but they might not have signed the authentication message yet.

It needs to be handled this way now because RainbowKit is handling our web3 connection state, and we're responding to changes in its internal state, rather than handling the connection ourselves. `isWalletConnected` is required to show the `Sign In` button (which triggers the authentication message signature).

Reply:
ok got it